bpi.syndicate.inc

```diff
@@ -332,149 +454,181 @@ function bpi_preprocess_bpi_search_results(&$variables) {
 }
 
 /**
- * Creates filters markup, based on current query string.
+ * Merge a bpi query into the current query string.
  *
- * @return string
- *   HTMl markup.
+ * @param array $query
+ *   The new query.
+ * @param bool|array $clear
+ *   Clear all or some parameters.
  *
- * @ingroup themeable
+ * @return array
+ *   The merged query.
  */
-function bpi_build_filters() {
-  $components = _bpi_build_query();
-  $filters_markup = array();
-  $enabled_filters = bpi_available_filters();
-
-  foreach ($components as $key => $value) {
-    // Limit the filtering options.
-    if (!in_array($key, $enabled_filters)) {
-      continue;
+function _bpi_build_query(array $query, $clear = FALSE) {
+  $current_query = $clear === TRUE ? array() : _bpi_get_query_parameter();
+  if (is_array($clear)) {
+    foreach ($clear as $name) {
+      unset($current_query[$name]);
     }
+  }
 
-    // Used in link.
-    $value_text = $value;
+  $bpi_query = array_replace_recursive($current_query, $query);
+  return $bpi_query;
+}
 
-    // Get agency name from cache.
-    if ($key == 'agency') {
-      $cache = cache_get(BPI_AGENCY_CACHE);
-      if ($cache && isset($cache->data[$value])) {
-        $value_text = $cache->data[$value];
+/**
+ * Get bpi query parameter value from query string.
+ *
+ * @param string|array $path
+ *   The parameter name or a list of names for multidimensional values.
+ * @param mixed $default_value
+ *   The default value.
+ *
+ * @return mixed
+ *   The query parameter value or the default value if not found.
+ */
+function _bpi_get_query_parameter($path = NULL, $default_value = NULL) {
+  $data = drupal_get_query_parameters();
+  if ($path && !is_array($path)) {
+    $path = array($path);
+  }
+  if ($path) {
+    foreach ($path as $step) {
+      if (!isset($data[$step])) {
+        return $default_value;
       }
+      $data = $data[$step];
     }
-
-    $item['name'] = $key;
-    $item['label'] = bpi_label_mapper($key);
-    $item['value'] = $value;
-    $item['search_value'] = $value;
-    $item['search_stripped_filter'] = _bpi_build_query('', '', $key);
-
-    $item['link'] = l($value_text, 'admin/bpi', array('query' => array($key => $value)));
-    $item['remove'] = l(
-      theme_image(
-        array(
-          'path' => drupal_get_path('module', 'bpi') . '/images/cross.png',
-          'width' => 16,
-          'height' => 16,
-          'alt' => '',
-          'attributes' => array(),
-        )
-      ),
-      'admin/bpi',
-      array(
-        'query' => $item['search_stripped_filter'],
-        'html' => TRUE,
-        'attributes' => array(
-          'class' => array('bpi-filter-remove-cross'),
-        ),
-      )
-    );
-
-    $filters_markup[] = theme('bpi_filter_item', array('filter' => $item));
   }
+  return $data;
+}
 
-  return implode('', $filters_markup);
+/**
+ * Used to retrieve the searched phrase after submit is clicked.
+ *
+ * @see bpi_syndicate_page_form()
+ *
+ * @return string
```

Comment:
Return type should be mixed, as the call to _bpi_get_query_parameter(). You could also set a default value as parameter 2.
 _bpi_get_query_parameter(BPI_SEARCH_PHRASE_KEY, FALSE);